src/jit/flowgraph.cpp

```diff
@@ -20588,6 +20588,270 @@ void Compiler::fgStress64RsltMul()
     fgWalkAllTreesPre(fgStress64RsltMulCB, (void*)this);
 }
 
+// BBPredsChecker checks jumps from the block's predecessors to the block.
+class BBPredsChecker
+{
+public:
+    BBPredsChecker(Compiler* compiler);
+
+    unsigned CheckBBPreds(BasicBlock* block, unsigned curTraversalStamp);
+
+private:
+    bool CheckEhTryDsc(BasicBlock* block, BasicBlock* blockPred, EHblkDsc* ehTryDsc);
+    bool CheckEhHndDsc(BasicBlock* block, BasicBlock* blockPred, EHblkDsc* ehHndlDsc);
+    bool CheckJump(BasicBlock* blockPred, BasicBlock* block);
+    bool CheckEHFinalyRet(BasicBlock* blockPred, BasicBlock* block);
+
+private:
+    Compiler* comp;
```

Comment:
Not sure if there's a rule for this but it's IMO preferable to put data members at the top of the class, if only because that's also where the constructor is so it's easier to see how and if data members are initialized.

Reply:
I would tend to agree with @mikedn on this, but the coding conventions say otherwise: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#13-c-classes. I'm pretty sure that there's very little consistency at this point in the current codebase. @BruceForstall - thoughts?

Reply:
I think this is where "C++" and "C#" coding conventions conflict.

The .NET coding convention have typically been "fields", "constructors", "properties", "methods" (with the old style-cop rules having more explicit ordering on top of this).

However, most C++ code follows something akin to what @CarolEidt linked above (Google's C++ coding style is similar: https://google.github.io/styleguide/cppguide.html#Declaration_Order).

I think the C#/.NET convention is much better organized and makes it significantly easier to find code in a file.